src/kvirc/kernel/KviDbusAdaptor.cpp

```diff
@@ -24,11 +24,37 @@
 
 #include "KviDbusAdaptor.h"
 
+#include "KviApplication.h"
+
 #ifdef COMPILE_DBUS_SUPPORT
 
+#include <QtDebug> //for qWarning()
+#include <QDBusConnection>
+
 KviDbusAdaptor::KviDbusAdaptor(QObject * pObj)
     : QDBusAbstractAdaptor(pObj)
 {
+    setAutoRelaySignals(false);
 }
 
+void KviDbusAdaptor::registerToSessionBus() {
+    auto connection = QDBusConnection::sessionBus();
+    if(!connection.registerService(KVI_DBUS_INTERFACENAME)) {
+        qWarning() << "D-Bus service registration failed:" << connection.lastError().message();
+    }
+    if(!connection.registerObject(KVI_DBUS_PATH, this, QDBusConnection::ExportAllSlots)) {
+        qWarning() << "D-Bus object registration failed:" << connection.lastError().message();
+    }
+}
+
+#ifndef COMPILE_NO_IPC
+void KviDbusAdaptor::ipcMessage(const QString &message)
+{
+    if(!g_pApp)
+        return;
+
+    g_pApp->ipcMessage(message.toUtf8().data());
```

Comment:
I'd prefer if the dbus interface was more structured than a random binary blob which might or might not be a KVS string

Reply:
And it doesn't have any way to return values

Reply:
I've created #2632 to keep track of it